### ISSUE TITLE
[prefer-destructuring] Variable declaration type annotation

### ISSUE THREAD

**Issue by ark120202** [typescript-eslint/typescript-eslint#723]
```ts
const foo = { bar: 1 as const };
const bar: number = foo.bar;
// Currently fixed to
const { bar } = foo;
```
1. Fixer should change code to `const { bar }: { bar: number } = foo`, removing type annotation is unsafe
1. I think it would make sense to add an option to disable rule in this case, annotations on destructured variables aren't very pretty

**Comment by armano2** [comment]
[Playground](https://typescript-eslint.io/play#ts=4.7.2&sourceType=module&code=MYewdgzgLgBAZiEMC8MDeMBGBDATgLhgEYZsIZRJYBfAbgFgAoS6LPQsAVwFtMBTXCniIAdDlwNGAeikwAwp1y4+YKABsAnvACWADz4ATGFBBMWsDOJjUhCELSA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQA7UQDNFoBaAExQBc4BjK2aASwDsBzdKE6Ae2knBgAviCFA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA)

### Code

```ts
const foo = { bar: 1 as const };
const bar: number = foo.bar;
// Currently fixed to
const { bar } = foo;
```

### Eslint config

```json
{
  "rules": {
    "prefer-destructuring": "error"
  }
}
```

**Comment by HolgerJeromin** [comment]
Can this issue be closed? ref PR #7117
